ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DirectionsQuestionnaireMapper.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.hmcts.cmc.ccd.mapper;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.ccd.domain.CCDYesNoOption;
+import uk.gov.hmcts.cmc.ccd.domain.directionsquestionnaire.CCDDirectionsQuestionnaire;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.DirectionsQuestionnaire;
+
+import java.util.stream.Collectors;
+
+@Component
+public class DirectionsQuestionnaireMapper implements Mapper<CCDDirectionsQuestionnaire, DirectionsQuestionnaire> {
+
+    private final ExpertRowMapper expertRowMapper;
+    private final UnavailableDateMapper unavailableDateMapper;
+
+    @Autowired
+    public DirectionsQuestionnaireMapper(ExpertRowMapper expertRowMapper, UnavailableDateMapper unavailableDateMapper) {
+        this.expertRowMapper = expertRowMapper;
+        this.unavailableDateMapper = unavailableDateMapper;
+    }
+
+    @Override
+    public CCDDirectionsQuestionnaire to(DirectionsQuestionnaire directionsQuestionnaire) {
+        if (directionsQuestionnaire == null) {
+            return null;
+        }
+
+        return CCDDirectionsQuestionnaire.builder()
+            .selfWitness(CCDYesNoOption.valueOf(directionsQuestionnaire.isSelfWitness()))
+            .howManyOtherWitness(directionsQuestionnaire.getHowManyOtherWitness())
+            .hearingLocation(directionsQuestionnaire.getHearingLocation())
+            .hearingLocationSlug(directionsQuestionnaire.getHearingLocationSlug())
+            .exceptionalCircumstancesReason(directionsQuestionnaire.getExceptionalCircumstancesReason())
+            .unavailableDates(directionsQuestionnaire.getUnavailableDates()
+                .stream()
+                .map(unavailableDateMapper::to)
+                .collect(Collectors.toList()))
+            .availableDate(directionsQuestionnaire.getAvailableDate())
+            .languageInterpreted(directionsQuestionnaire.getLanguageInterpreted())
+            .signLanguageInterpreted(directionsQuestionnaire.getSignLanguageInterpreted())
+            .hearingLoop(CCDYesNoOption.valueOf(directionsQuestionnaire.isHearingLoop()))
+            .disabledAccess(CCDYesNoOption.valueOf(directionsQuestionnaire.isDisabledAccess()))
+            .otherSupportRequired(directionsQuestionnaire.getOtherSupportRequired())
+            .expertEvidenceToExamine(directionsQuestionnaire.getExpertEvidenceToExamine())
+            .expertReportsRows(directionsQuestionnaire.getExpertReportsRows()
+                .stream()
```

Comment:
Please use `StreamUtil.asStream` utility method which is null safe

Reply:
done